linode/manager
# upcoming: [M3-7993] - PlacementGroups Select optimizations & cleanup

## Description 📝
This PR addresses small tech debt left by iterations on the PlacementGroupSelect component and related utils.

## Changes  🔄
- `PlacementGroupSelect` now filters PGs via query instead of client-side
- Cleanup `PlacementGroupSelect` props and remove what's unused
- Remove duplicate state management in `PlacementGroupDetailPanel`
- Use `selectedPlacementGroupId` rather than the whole PG object for prop drilling
- Allow `useAllPlacementGroupsQuery` query to take a filter
- Small capitalization copy change for the `PlacementGroupSelect`  `noOptionsMessage`

## Preview 📷
There should be no visual change as a result of this PR

## How to test 🧪
ℹ️ Test in ALPHA

### Verification steps
- Confirm behavior of PG select in Linode Create Flow (both v1 and v2)
  - Make sure PGs are properly filtered by region
  - Make sure flows work and PG is selected in the final payload
  - Make sure we can create & select a new PG in the the v1 and v2 linode create flows
- Confirm behavior of PG select in Linode Migration Flow

## As an Author I have considered 🤔

*Check all that apply*
